### PR TITLE
tests(web-inspector): add test for setImmediate polyfill

### PR DESCRIPTION
**Pull request by patrickhulce**
reminded by gitter to add test for this :)

### CODE REVIEW

File: lighthouse-core/test/lib/web-inspector-test.js [hunk 1]
```diff
@@ -21,6 +21,21 @@ describe('Web Inspector lib', function() {
     assert.ok(WebInspector.Color);
   });
 
+  it('does not destroy setImmediate', done => {
```
Comment (brendankenny) [inline lighthouse-core/test/lib/web-inspector-test.js]: not immediately clear what's being tested here. Is this checking for the specific WebInspector (bad) `setImmediate` implementation?

Comment (patrickhulce) [reply lighthouse-core/test/lib/web-inspector-test.js]: yes

`does not break setImmediate`
`does not polyfill setImmediate incorrectly`

any better?

Comment (brendankenny) [reply lighthouse-core/test/lib/web-inspector-test.js]: yeah, that helps, though maybe a comment about why `arg.foo();` would throw, too :)

Comment (brendankenny) [reply lighthouse-core/test/lib/web-inspector-test.js]: also TIL `setImmediate` also exists on `require('timers').setImmediate`, so could check `setImmediate === require('timers').setImmediate` in addition to this functional test.

In theory someone could overwrite that as well, but they may deserve to screw things up by the time they reach that level of effort.

**Pull request merged**